src/github.com/getlantern/lantern-mobile/app/src/main/java/org/getlantern/lantern/fragment/FeedFragment.java

```diff
@@ -62,28 +55,48 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         View view = inflater.inflate(R.layout.feed_fragment, container, false);
 
         mList = (ListView)view.findViewById(R.id.feed);
-        adapter = new FeedAdapter(getActivity(), mFeedItems);
         mList.setAdapter(adapter);
         return view;
     }
 
+    protected class LoadFeed extends AsyncTask<String, Void, List<FeedItem>> {
+
+        @Override
+        protected List<FeedItem> doInBackground(String... params) {
+
+            String name = params[0];
+
+            final List<FeedItem> items = new ArrayList<FeedItem>();
+
+            Lantern.FeedByName(name, new Lantern.FeedRetriever.Stub() {
+                public void AddFeed(String title, String desc, 
+                        String image, String url) {
+                    items.add(new FeedItem(title, desc, image, url));
+                }
+            });
+
+            return items;
+        }
+
+        @Override
+        protected void onPostExecute(List<FeedItem> items) {
+            super.onPostExecute(items);
+            mFeedItems.clear();
```

Comment:
It doesn't need to be synchronized any longer (that's why I moved the processing inside an AsyncTask). It happens on the UI thread now.